functions/bfc-functions.php

```diff
@@ -124,6 +124,121 @@ function bfc_nouveau_has_nav( $args = array() ) {
 	return true;
 }
 
+/* Main Nav Menu */
+function bfc_top_nav_menu_builder() {
+$menuitemarray = array (
+  array (
+    'bfc-topnav-home', // menu item id (and selector name)
+    'menu-item-home menu-item-type-custom menu-item-object-custom current-menu-item current_page_item', // any unique selectors for menu item
+    'home', // parent top nav to highlight
+    '/', // page, use relative urls
+    '/wp-content/themes/bfcom/assets/images/home.svg', // icon          // menu item icon
+    'Home' // menu item text
```

Comment:
I love that you're storing this stuff and an array; keeps the data clean by not mixing it directly into the code. I also appreciate the comments to show what each entry is for.

I think even better would be to define explicit keys for each item in the array (thus making it an associative array), like so:

```
array (
  'id' => 'bfc-topnav-home', // menu item id (and selector name)
  'classes' => 'menu-item-home menu-item-type-custom menu-item-object-custom current-menu-item current_page_item', // any unique selectors for menu item
  'parent_nav' => 'home', // parent top nav to highlight
  'link_url' => '/', // page, use relative urls
  'icon_url' => '/wp-content/themes/bfcom/assets/images/home.svg', // icon          // menu item icon
  'text' => 'Home' // menu item text
),
```

Then you can refer to them later in a more meaningful way:

```
$thismenuitem .= $menuitem['id'];
```

For more information, see:

http://php.net/manual/en/language.types.array.php

Reply:
Fixed. I was considering using key-value pairs. I think it's a good idea since we're likely to add to these arrays and this will be far easier to extend. It also improves clarity so much that I removed the inline comments that seemed redundant.